pdns/recursordist/rec-zonetocache.cc

```diff
@@ -0,0 +1,288 @@
+/*
+ * This file is part of PowerDNS or dnsdist.
+ * Copyright -- PowerDNS.COM B.V. and its contributors
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * In addition, for the avoidance of any doubt, permission is granted to
+ * link this program with OpenSSL and to (re)distribute the binaries
+ * produced as the result of such linking.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "rec-zonetocache.hh"
+
+#include "syncres.hh"
+#include "zoneparser-tng.hh"
+#include "query-local-address.hh"
+#include "axfr-retriever.hh"
+#include "validate-recursor.hh"
+#include "logging.hh"
+#include "threadname.hh"
+#include "rec-lua-conf.hh"
+
+#ifdef HAVE_LIBCURL
+#include "minicurl.hh"
+#endif
+
+#include <fstream>
+
+struct ZoneData
+{
+  ZoneData(shared_ptr<Logr::Logger>& log) :
+    d_log(log) {}
+
+  std::map<pair<DNSName, QType>, vector<DNSRecord>> d_all;
+  std::map<pair<DNSName, QType>, vector<shared_ptr<RRSIGRecordContent>>> d_sigs;
```

Comment:
Would it make sense to merge these two maps into one?

Reply:
It's not clear to me if that wild make the code easier to understand. I'll add a comment in the code that it *might* be an improvement.

Reply:
Works for me! I was under the impression that we usually use the same key and access both maps, but I realize that this is not 100% true for RRSIGs.